README.md

```diff
@@ -242,6 +243,85 @@ example: array.first('array'), // '1'
 composingExample: array.first(split('string', raw(', '))) // '1'
 ```
 
+##### `array.groupBy`
+allows to group an array of objects by key with an optional comparator
+
+```js
+array: Ember.A([{ test: 1, name: 'foo'}, { test: 2, name: 'foo' }, { test: 1, name: 'bar' }]),
+key: 'test',
+value: array.groupBy('array', 'key')
+/*
```

Comment:
It is a convention that I made up.
We can make it lodash alike, though. Given the example in the readme, the result would be
```js
{ 
    1: [{ test: 1 , name: 'foo' }, { test: 1 , name: 'bar' }] },
    2: [{ test: 2 , name: 'foo' }]
}
```
in which the object property name equals the value of the attribute used to group. In this case a Number, but it could be a String.

However I think it's a major drawback because:
1. If I want to group by a complex object like Date, that object cannot be used as property name inside the resulting object
2. The result becomes an object and not an array. 
    - I think that for the sake of consistency and chaining if we start from an array of objects and we use a computed macro belonging to the array module we should return at least a simple array, or better an array of objects
    - Not user friendly in term of looping